src/actions/mod.rs

```diff
@@ -33,7 +34,63 @@ use std::time::Duration;
 
 use self::compiler_message_parsing::{FileDiagnostic, ParseError};
 
-type BuildResults = HashMap<PathBuf, Vec<Diagnostic>>;
+type BuildResults = HashMap<PathBuf, Vec<RustDiagnostic>>;
+
+#[derive(Debug, PartialEq, Deserialize, Serialize)]
```

Comment:
I think it would be nice to have a comment saying why these exist and how they are different from the LSP